tests/tests.js

```diff
@@ -1,14 +1,79 @@
 const tape = require('tape');
-const request = require('supertest');
+const supertest = require('supertest');
 const server = require('../server.js');
+const axios = require('axios');
+const jwt = require('jsonwebtoken');
+require('dotenv').config();
+
+const postPlace = require('../routes/middleware/post_place.js');
 
 tape('home route test: GET request to /', t => {
-  request(server)
+  supertest(server)
     .get('/')
     .end((err, res) => {
-      if (err) throw err;
+      t.error(err);
       t.equal(res.status, 200, 'should return status code 200');
-      t.ok(res.text.includes('<div id="list-page-content">'), 'should return home page containing test string "<div id="list-page-content">"');
+      t.ok(res.text.includes('<div id="list-page-content">'));
+      t.end();
+    });
+});
+
+tape('Test the event details route', async t => {
+  let eventId = '';
+  try {
+    const eventResponse = await axios.get(
+      'https://nazareth-open-tourism-platform.herokuapp.com/api/v1/events'
+    );
+    const parsedBody = eventResponse.data;
+    const lastIndex = parsedBody.length - 1;
+    const htmlSample = `<h1 class="event-name-details">${
+      parsedBody[lastIndex].en.name
+    }</h1>`;
+    eventId = parsedBody[lastIndex]._id;
+    supertest(server)
+      .get('/en/events/' + eventId)
+      .end((err, res) => {
+        t.error(err);
+        t.ok(res.text.includes(htmlSample), 'It found the right event');
+      });
+    supertest(server)
+      .get('/en/events/someid')
+      .end((err, res) => {
+        t.error(err);
+        t.ok(res.text.includes('<title>Not Found</title>'));
+        t.end();
+      });
+  } catch (err) {
+    console.log(err);
```

Comment:
Try catch shouldn't really be used within tests. If you want to test if something within your tests throws an error, use [t.throws](https://github.com/substack/tape#tthrowsfn-expected-msg) or similar

Reply:
Also `console.log`s aren't really tests...

Reply:
yeah forgot about that good spot :+1: